src/hooks/api/useCashFlow.ts

```diff
@@ -46,12 +46,14 @@ export function useCashFlow(
             ) AS mnemonic,
             SUM(
               CASE
-                WHEN splits.guid = credit_split.guid THEN
+                WHEN splits.guid = asset_split.guid THEN
```

Comment:
credit_split was incorrect because for income transactions this means we are selecting the wrong account. Changed the logic to select asset type account (except INVESTMENT) which keeps all the tests passing and also passes the newly added one (plus correct behavior). Whenever we discover a new edge case can always update